tests/testthat/test-merge_sce_list.R

```diff
@@ -578,17 +578,187 @@ test_that("merging SCEs where 1 altExp is missing works as expected, with altexp
 
   # check that the NAs are as expected
   counts_mat <- counts(merged_altexp)
-  sce4_counts <- counts_mat[, merged_sce$library_id == "sce4"]
+  sce4_counts <- counts_mat[, merged_sce[[batch_column]] == "sce4"]
   expect_true(
     all(is.na(sce4_counts))
   )
-  numeric_counts <- counts_mat[, merged_sce$library_id != "sce4"]
+  numeric_counts <- counts_mat[, merged_sce[[batch_column]] != "sce4"]
   expect_true(
     all(is.finite(numeric_counts))
   )
 })
 
 
+test_that("merging SCEs with different altExps works as expected; each SCE has 1 altExp of a different name", {
+  sce1 <- sce_list_with_altexp[[1]]
+  sce2 <- sce_list_with_altexp[[2]]
+  other_altexp_name <- "other"
+  altExpNames(sce2) <- other_altexp_name
+  rownames(altExp(sce2)) <- c("ADT-A", "ADT-B", "ADT-C", "ADT-D", "ADT-E")
+
+  sce_list <- list(
+    "sce1" = sce1,
+    "sce2" = sce2
+  )
+
+  # Merge:
+  merged_sce <- merge_sce_list(
+    sce_list,
+    batch_column = batch_column,
+    # "total" should get removed
+    retain_coldata_cols = retain_coldata_cols,
+    # this row name should not be modified:
+    preserve_rowdata_cols = c("gene_names")
+  )
+
+  # Correct names
+  expect_equal(
```

Comment:
since we don't care about order here
```suggestion
  expect_setequal(
```